src/Fmt_ast.ml

```diff
@@ -2685,6 +2687,11 @@ and fmt_cases c ctx cs =
                   (fmt_or c.conf.indicate_multiline_delimiters "@ )" "@,)")
               ) ) )
 
+and is_arrow_or_poly = function
```

Comment:
This does not need to be mutually recursive. Could you move it to the top of the file ?